### ISSUE TITLE
Filters - email removes the ' character

### ISSUE THREAD

**Issue by AmazingDreams** [phalcon/cphalcon#10595]
According to Wikipedia the following special characters are allowed in an email address:
```
"These special characters: # - _ ~ ! $ & ' ( ) * + , ; = : and percentile encoding i.e. %20"
```

https://en.wikipedia.org/wiki/Email_address#Local_part

Also when you do a 'native' php filter_var call, the `'` character is not removed:

```
$email = "some'email@example.com";
echo filter_var($email, FILTER_SANITIZE_EMAIL); // some'email@example.com
```

Why does Phalcon remove the `'` character?

https://github.com/phalcon/cphalcon/blob/master/phalcon/filter.zep#L150

**Comment by williamokano** [comment]
Good question. Apostrophes should be valid before the @ symbol yet most of the big e-mail companies do not accept them.

:+1:

**Comment by AmazingDreams** [comment]
I'd say that following the standards is more important than following big e-mail companies. This gives Phalcon users more fine-grained control over what they want to allow. I was pretty surprised myself that the `'` got removed.

Besides, if you look at the email validator, it uses the native `FILTER_VALIDATE_EMAIL` constant. https://github.com/phalcon/cphalcon/blob/master/phalcon/validation/validator/email.zep#L55

**Comment by AmazingDreams** [comment]
I closed the old pull request as that one was trying to merge with master. Which I shouldnt

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
